plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionFields.java

```diff
@@ -29,18 +29,21 @@
 
 public final class PartitionFields
 {
-    private static final String NAME = "[a-z_][a-z0-9_]*";
-    private static final String FUNCTION_ARGUMENT_NAME = "\\((" + NAME + ")\\)";
-    private static final String FUNCTION_ARGUMENT_NAME_AND_INT = "\\((" + NAME + "), *(\\d+)\\)";
+    private static final String IDENTIFIER = "[a-z_][a-z0-9_]*";
+    private static final String QUOTED_IDENTIFIER = "\"[^\"]*(?:(?:\"\")+[^\"]*)*\"";
```

Comment:
Can you add a comment for each of the non-trivial regex patterns?

Reply:
I think with @findepi's great suggestion, that regex is now a lot simpler. It was indeed a bit convoluted.